# Module 1 Lab 3: Choosing an Insurance Provider

## Problem 1.2

This analysis compares three insurance providers using their customer ratings. I will consider the minimum rating, mean rating, and maximum rating to make a recommendation while considering both typical performance and worst-case risk.

In [1]:
# Customer ratings for each insurance provider

geico = [4.7, 8.3, 9.2]
progressive = [7.4, 6.7, 8.9]
usaa = [3.8, 6.3, 8.1]

print("GEICO ratings:", geico)
print("Progressive ratings:", progressive)
print("USAA ratings:", usaa)

GEICO ratings: [4.7, 8.3, 9.2]
Progressive ratings: [7.4, 6.7, 8.9]
USAA ratings: [3.8, 6.3, 8.1]


In [2]:
# Calculate the minimum, mean, and maximum rating for each provider

providers = {
    "GEICO": geico,
    "Progressive": progressive,
    "USAA": usaa
}

for provider, ratings in providers.items():
    print(provider)
    print("Minimum:", min(ratings))
    print("Mean:", round(sum(ratings) / len(ratings), 2))
    print("Maximum:", max(ratings))
    print()

GEICO
Minimum: 4.7
Mean: 7.4
Maximum: 9.2

Progressive
Minimum: 6.7
Mean: 7.67
Maximum: 8.9

USAA
Minimum: 3.8
Mean: 6.07
Maximum: 8.1



## Summary Table

| Insurance Provider | Minimum Rating | Mean Rating | Maximum Rating |
|---|---:|---:|---:|
| GEICO | 4.7 | 7.40 | 9.2 |
| Progressive | 6.7 | 7.67 | 8.9 |
| USAA | 3.8 | 6.07 | 8.1 |

The mean shows the typical customer rating, while the minimum shows the worst rating in the available data. Looking at both measures helps avoid choosing a provider based only on the average.

In [3]:
# Identify the provider with the highest mean rating and highest minimum rating

highest_mean = max(providers, key=lambda provider: sum(providers[provider]) / len(providers[provider]))
highest_minimum = max(providers, key=lambda provider: min(providers[provider]))

print("Highest mean rating:", highest_mean)
print("Highest minimum rating:", highest_minimum)

Highest mean rating: Progressive
Highest minimum rating: Progressive


## Recommendation

I would prefer Progressive as the insurance provider. It has the highest average rating of about 7.67, which implies that customers had the best experience on average compared to the other two providers. More importantly, it also has the highest minimum rating of 6.7, which means that its worst-case rating is still higher than the worst-case rating of GEICO or USAA. GEICO has a slightly higher maximum rating of 9.2, but its minimum rating is only 4.7, which is much worse and implies more risk of a bad customer experience. Finally, USAA has the lowest minimum rating of 3.8 and the lowest average rating of about 6.07. Thus, based on the analysis, Progressive appears to be the best choice.

## Conclusion

Based on the available ratings, Progressive is the strongest overall choice because it combines the highest average rating with the highest minimum rating. The decision considers both the typical customer experience and the possibility of a poor experience rather than relying only on the average.

## GitHub Repository

Repository URL: **Add your GitHub portfolio repository URL here before submitting.**

Save this notebook as `module1_lab3.ipynb` and place it in the `module1/` folder of the GitHub portfolio repository.